### PR TITLE
Change ``NoReturn`` to be encapsulated in strings

### LINKED ISSUE #1239: TypeError: Plain typing.NoReturn is not valid as type argument in Astroid 2.8.5 + Python 3.7.0/1
### Steps to reproduce

1. Use Python 3.7.0 or 3.7.1
2. pip install astroid==2.8.5
3. python -c "import astroid"

### Current behavior

An error is raised
```
TypeError: Plain typing.NoReturn is not valid as type argument
```

In this commit https://github.com/PyCQA/astroid/commit/b62f243b16b7f435c8be869577959e95a7927a91#diff-039e09546480e9ce70fc9a23e975f1ac2364f6d0192e330311d204653947a216R34 there is a conditional import that checks if Python >= 3.6.2, but that should also include Python 3.7.0 and 3.7.1 (which have the same issue as Python 3.6.0/1).

### Expected behavior

There should be no error

### python -c "from astroid import __pkginfo__; print(__pkginfo__.version)" output

This raises the same error as above

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description

Follow up to #1235

Closes #1239

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -12,7 +12,7 @@ What's New in astroid 2.8.6?
 ============================
 Release date: TBA
 
-
+* Fix bug on Python 3.7.0 and 3.7.1 by removing unallowed usage of ``typing.NoReturn``
```
Comment (cdce8p) [inline ChangeLog:15]:
```suggestion
* Fix bug with Python 3.7.0 / 3.7.1 and ``typing.NoReturn``.

   Closes #1239
```
Maybe something like this?

File: ChangeLog [hunk 2]
```diff
@@ -12,7 +12,9 @@ What's New in astroid 2.8.6?
 ============================
 Release date: TBA
 
+* Fix bug with Python 3.7.0 / 3.7.1 and ``typing.NoReturn``.
 
+   Closes #1239
```
Comment (cdce8p) [inline ChangeLog:17]:
```suggestion
  Closes #1239
  
```

**Review by cdce8p — APPROVED** [review]
Thanks @DanielNoord 🐬

**Review by Pierre-Sassoulas — APPROVED** [review]
🎉

**Pull request merged**